src/Microsoft.ML.Data/Model/ModelLoading.cs

```diff
@@ -134,8 +179,6 @@ public bool ContainsModel(string name)
                     return true;
             }
 
-            // TryLoadModelCore should rewind on failure.
-            Contracts.Assert(fp == ent.Stream.Position);
```

Comment:
>Contracts.Assert(fp == ent.Stream.Position); [](start = 12, length = 44)

The first thing to look at on this PR is whether the changes on removing throwing for the case when the header length/body length of the expected signature, and the actual signature in the TryLoadModel is acceptable. 

I feel like it can be ok to not throw, since this is a Try-type method, but i know that this area of the codebase has been hashed pretty well in the past :)

If we are not ok with deleting this assert, and the ones below in TryLoadModelCore, and we also want to preserve backwards compatibility, I'll unwind the changes to have all the stats in one class. (and with that have methods like GetBais, or GetCoefficients exit when the std of the coefficients is not provided.. ) 

If we don't care about backwards compatibility, we can keep those asserts.